proxy/http2/unit_tests/test_HTTP2.cc

```diff
@@ -107,8 +107,8 @@ TEST_CASE("Convert HTTPHdr", "[HTTP2]")
 
     // check
     CHECK_THAT(buf, Catch::StartsWith("GET https://trafficserver.apache.org/index.html HTTP/1.1\r\n"
-                                      "Host: trafficserver.apache.org\r\n"
                                       "User-Agent: foobar\r\n"
+                                      "Host: trafficserver.apache.org\r\n"
```

Comment:
This is not great.

> A user agent that sends Host SHOULD send it as the first field in the header section of a request.

https://datatracker.ietf.org/doc/html/rfc9110#name-host-and-authority

Reply:
I see. This doesn't look trivial to fix.

When converting from h2 to h1, we take the `:authority` header and get a host value out of it and append a `Host:` header for the resultant HTTP/1 creation. Our `field_create` function appends and there is no corresponding function to prepend a header. Do you know a way offhand to prepend a header field at the beginning?

Reply:
Ah now I understand what you mean. We know there's ":authority" pseudo header before normal headers (if the request header is valid). How about renaming the field to Host?

Reply:
This is a creative idea.

This actually turned out to be more challenging than I expected, though. Setting a field's name requires that the field be in the "detached" rather than the "live" state. At this point in the conversion, we are modifying the HTTPHdr and the fields are all live. I was able to avoid an append by first detaching the field, modifying the value, then re-attaching (which preserved the original order). I'm not sure what the performance implications of the detach operation are, however. It does look like the results are now functionally correct, however. The `:authority` header, with this latest patch, is now transformed into a `Host:` header and that `Host:` header is at the start of the header block after the request list.

Let me know whether you have any concerns with this.

Reply:
Detaching field should not be that heavy. It's just changing a couple of flags, IIUC.